log4j-to-jul/src/main/java/org/apache/logging/log4j/tojul/JULLogger.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.apache.logging.log4j.tojul;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.message.MessageFactory;
+import org.apache.logging.log4j.spi.AbstractLogger;
+
+/**
+ * Implementation of {@link org.apache.logging.log4j.Logger} that's backed by a {@link Logger}.
+ *
+ * This implementation currently ignores {@link Marker}.
+ *
+ * @author <a href="http://www.vorburger.ch">Michael Vorburger.ch</a> for Google
+ */
+public class JULLogger extends AbstractLogger {
```

Comment:
My preference is final-by-default for new types

Reply:
Sure, done. May I ask why, just for learning? Does this make a (real, noticeable, measurable) difference for performance? Or is it not for performance but "style" - you don't want classes to be "open" and extensible by users? Thanks!

Reply:
Of course :-)
The final class means we don't have to design or review for all the ways an external consumer may decide to extend this class. This way we don't need to review as many edge cases, and we have much greater flexibility to change the class in the future without causing odd semantic breaks.

I don't believe there's any measurable performance impact (C2 should be able to optimize non-final and final methods  similarly if the receiver type is always the same. Also, bytecode manipulation allows final classes to be subclassed, e.g. Mockito + inline mock maker)

Reply:
Thanks! Resolving.